include/swift/AST/ClangModuleLoader.h

```diff
@@ -100,6 +101,17 @@ class ClangModuleLoader : public ModuleLoader {
   lookupRelatedEntity(StringRef clangName, ClangTypeKind kind,
                       StringRef relatedEntityKind,
                       llvm::function_ref<void(TypeDecl *)> receiver) = 0;
+
+  /// Try to parse the string as a Clang function type.
+  ///
+  /// Returns null if there was a parsing failure.
+  virtual const clang::Type *parseClangFunctionType(StringRef type,
+                                                    StringRef filename,
```

Comment:
The filename should always be recoverable from the `SourceLoc`.

Reply:
Oh right, that is what I'm doing in the caller. D'oh!

Reply:
Fixed.